### PR TITLE
Refactor login info

### PR DESCRIPTION
**Pull request by omgronny**

### CODE REVIEW

File: src/main/java/org/itmo/eventapp/main/model/entity/UserLoginInfo.java [hunk 1]
```diff
@@ -24,15 +24,19 @@ public class UserLoginInfo implements UserDetails {
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Integer id;
 
-    @OneToOne
+    @OneToOne(fetch = FetchType.EAGER)
```
Comment (HackMemory) [inline src/main/java/org/itmo/eventapp/main/model/entity/UserLoginInfo.java]: в итоге оставляем one to one?

File: src/main/java/org/itmo/eventapp/main/security/service/UserDetailsServiceImpl.java [hunk 2]
```diff
@@ -15,8 +15,7 @@ public class UserDetailsServiceImpl implements UserDetailsService {
 
     @Override
     public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
-
-        return userLoginInfoRepository.getUserLoginInfoByEmail(username)
+        return userLoginInfoRepository.getUserLoginInfoByLogin(username)
                 .orElseThrow(() -> new UsernameNotFoundException("No user with email: " + username));
```
Comment (HackMemory) [inline src/main/java/org/itmo/eventapp/main/security/service/UserDetailsServiceImpl.java:19]: тут видимо нужно поправить еще сообщение об ошибке

**Comment by IP-13** [comment]
|Overall Project|71.62% **`-0.14%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|93.68%|:green_apple:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[TaskNotificationUtils.java](https://github.com/itmo-events-app/event-app/blob/bb70b0ed9f671af0e558c3080db92e0ff25e5ba4/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Futil%2FTaskNotificationUtils.java)|94.74%|:green_apple:|
|[JwtTokenUtil.java](https://github.com/itmo-events-app/event-app/blob/bb70b0ed9f671af0e558c3080db92e0ff25e5ba4/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fsecurity%2Futil%2FJwtTokenUtil.java)|90.7%|:green_apple:|
|[JwtFilter.java](https://github.com/itmo-events-app/event-app/blob/bb70b0ed9f671af0e558c3080db92e0ff25e5ba4/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fsecurity%2Ffilter%2FJwtFilter.java)|90%|:green_apple:|
|[TaskService.java](https://github.com/itmo-events-app/event-app/blob/bb70b0ed9f671af0e558c3080db92e0ff25e5ba4/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FTaskService.java)|89.09%|:green_apple:|
|[TaskController.java](https://github.com/itmo-events-app/event-app/blob/bb70b0ed9f671af0e558c3080db92e0ff25e5ba4/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FTaskController.java)|87.62%|:green_apple:|
|[UserService.java](https://github.com/itmo-events-app/event-app/blob/bb70b0ed9f671af0e558c3080db92e0ff25e5ba4/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserService.java)|86.72%|:green_apple:|
|[UserLoginInfoService.java](https://github.com/itmo-events-app/event-app/blob/bb70b0ed9f671af0e558c3080db92e0ff25e5ba4/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserLoginInfoService.java)|81.25%|:green_apple:|
|[UserDetailsServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/bb70b0ed9f671af0e558c3080db92e0ff25e5ba4/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fsecurity%2Fservice%2FUserDetailsServiceImpl.java)|71.43%|:green_apple:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/bb70b0ed9f671af0e558c3080db92e0ff25e5ba4/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|31.67% **`-3.33%`**|:green_apple:|

File: src/main/java/org/itmo/eventapp/main/service/AuthenticationService.java [hunk 3]
```diff
@@ -53,7 +53,7 @@ public String login(LoginRequest loginRequest) {
     public void createRegisterRequest(RegistrationUserRequest registrationUserRequest) {
         String login = registrationUserRequest.email();
```
Comment (HackMemory) [inline src/main/java/org/itmo/eventapp/main/service/AuthenticationService.java]: тут остался email вроде

File: src/test/resources/db/test-migration/V0_2__create_tables_test.sql [hunk 4]
```diff
@@ -45,8 +45,9 @@ create table if not exists user_t
...
     id int generated always as identity primary key not null,
     user_id int not null references user_t(id),
-    email varchar(256) not null unique,
-    email_status email_status not null,
+    login varchar(256) not null unique,
+    login_status login_status not null,
+    login_type login_type not null,
     password_hash varchar(512) not null,
     reset_token varchar(512),
     last_login_date date,
```
Comment (HackMemory) [inline src/test/resources/db/test-migration/V0_2__create_tables_test.sql:53]: это все таки оставляем в userinfo или лучше перенести в user?

возникает проблема, что у разных логинов, будет разное время входа

File: src/main/java/org/itmo/eventapp/main/model/entity/UserLoginInfo.java (hunk 1 above)
Comment (omgronny) [reply src/main/java/org/itmo/eventapp/main/model/entity/UserLoginInfo.java]: Нет, это я для простоты этого конкретного ПРа сделал

File: src/main/java/org/itmo/eventapp/main/service/AuthenticationService.java (hunk 3 above)
Comment (omgronny) [reply src/main/java/org/itmo/eventapp/main/service/AuthenticationService.java]: Да, именно в этом был баг, уже поправил

File: src/test/resources/db/test-migration/V0_2__create_tables_test.sql (hunk 4 above)
Comment (HackMemory) [reply src/test/resources/db/test-migration/V0_2__create_tables_test.sql:53]: я про last_login_date

File: src/main/java/org/itmo/eventapp/main/security/service/UserDetailsServiceImpl.java (hunk 2 above)
Comment (omgronny) [reply src/main/java/org/itmo/eventapp/main/security/service/UserDetailsServiceImpl.java:19]: Это тоже не считаю необходимой частью этого ПРа
Он и так перегружен логикой в куче мест

Comment (omgronny) [reply src/main/java/org/itmo/eventapp/main/security/service/UserDetailsServiceImpl.java]: Хотя ладно, сделал. Одно слово поменять не страшно

**Comment by IP-13** [comment]
|Overall Project|71.62% **`-0.27%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|88.35%|:green_apple:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[TaskNotificationUtils.java](https://github.com/itmo-events-app/event-app/blob/dc70b0e64b2df999fb3e9b6fd7f249c5c3df83fa/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Futil%2FTaskNotificationUtils.java)|94.74%|:green_apple:|
|[JwtTokenUtil.java](https://github.com/itmo-events-app/event-app/blob/dc70b0e64b2df999fb3e9b6fd7f249c5c3df83fa/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fsecurity%2Futil%2FJwtTokenUtil.java)|90.7%|:green_apple:|
|[JwtFilter.java](https://github.com/itmo-events-app/event-app/blob/dc70b0e64b2df999fb3e9b6fd7f249c5c3df83fa/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fsecurity%2Ffilter%2FJwtFilter.java)|90%|:green_apple:|
|[TaskService.java](https://github.com/itmo-events-app/event-app/blob/dc70b0e64b2df999fb3e9b6fd7f249c5c3df83fa/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FTaskService.java)|89.09%|:green_apple:|
|[TaskController.java](https://github.com/itmo-events-app/event-app/blob/dc70b0e64b2df999fb3e9b6fd7f249c5c3df83fa/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FTaskController.java)|87.62%|:green_apple:|
|[UserService.java](https://github.com/itmo-events-app/event-app/blob/dc70b0e64b2df999fb3e9b6fd7f249c5c3df83fa/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserService.java)|86.72%|:green_apple:|
|[UserLoginInfoService.java](https://github.com/itmo-events-app/event-app/blob/dc70b0e64b2df999fb3e9b6fd7f249c5c3df83fa/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserLoginInfoService.java)|81.25%|:green_apple:|
|[UserDetailsServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/dc70b0e64b2df999fb3e9b6fd7f249c5c3df83fa/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fsecurity%2Fservice%2FUserDetailsServiceImpl.java)|71.43% **`-28.57%`**|:green_apple:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/dc70b0e64b2df999fb3e9b6fd7f249c5c3df83fa/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|31.67% **`-3.33%`**|:green_apple:|

**Review by HackMemory — APPROVED** [review]

File: src/main/java/org/itmo/eventapp/main/repository/UserRepository.java [hunk 5]
```diff
@@ -12,5 +12,5 @@
 public interface UserRepository extends JpaRepository<User, Integer> {
     List<User> findAllByRole(Role role);
 
-    Optional<User> findByUserLoginInfo_Email(String email);
+    Optional<User> findByUserLoginInfo_Login(String login);
```
Comment (IP-13) [inline src/main/java/org/itmo/eventapp/main/repository/UserRepository.java:15]: Убери тут сразу нижнее подчеркивание, оно не нужно

Comment (IP-13) [reply src/main/java/org/itmo/eventapp/main/repository/UserRepository.java:15]: Ваще не понял кстати откуда это, странно
Короче это надо удалить. По коду посмотрел, там в taskController используется только метод, в котором используется этот метод. Надо в taskController заменить userService.findByUserLoginInfoLogin на userLoginInfoService.findByEmail. Потому что этот метод супер запутанный, давай это замерджим, а в отдельном ПР-е исправим

**Review by IP-13 — APPROVED** [review]

**Comment by IP-13** [comment]
|Overall Project|72.31% **`-0.39%`**|:green_apple:|
|:-|:-|:-:|
|Files changed|85.12%|:green_apple:|
<br>

|File|Coverage||
|:-|:-|:-:|
|[TaskNotificationUtils.java](https://github.com/itmo-events-app/event-app/blob/b7302f43290357c755e29c7d9016017da4f40244/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Futil%2FTaskNotificationUtils.java)|94.74%|:green_apple:|
|[JwtTokenUtil.java](https://github.com/itmo-events-app/event-app/blob/b7302f43290357c755e29c7d9016017da4f40244/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fsecurity%2Futil%2FJwtTokenUtil.java)|90.7%|:green_apple:|
|[JwtFilter.java](https://github.com/itmo-events-app/event-app/blob/b7302f43290357c755e29c7d9016017da4f40244/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fsecurity%2Ffilter%2FJwtFilter.java)|90%|:green_apple:|
|[TaskService.java](https://github.com/itmo-events-app/event-app/blob/b7302f43290357c755e29c7d9016017da4f40244/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FTaskService.java)|88.55%|:green_apple:|
|[TaskController.java](https://github.com/itmo-events-app/event-app/blob/b7302f43290357c755e29c7d9016017da4f40244/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fcontroller%2FTaskController.java)|88.5% **`-3%`**|:green_apple:|
|[UserService.java](https://github.com/itmo-events-app/event-app/blob/b7302f43290357c755e29c7d9016017da4f40244/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserService.java)|86.72%|:green_apple:|
|[UserLoginInfoService.java](https://github.com/itmo-events-app/event-app/blob/b7302f43290357c755e29c7d9016017da4f40244/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FUserLoginInfoService.java)|81.25%|:green_apple:|
|[UserDetailsServiceImpl.java](https://github.com/itmo-events-app/event-app/blob/b7302f43290357c755e29c7d9016017da4f40244/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fsecurity%2Fservice%2FUserDetailsServiceImpl.java)|71.43% **`-28.57%`**|:green_apple:|
|[AuthenticationService.java](https://github.com/itmo-events-app/event-app/blob/b7302f43290357c755e29c7d9016017da4f40244/src%2Fmain%2Fjava%2Forg%2Fitmo%2Feventapp%2Fmain%2Fservice%2FAuthenticationService.java)|31.67% **`-3.33%`**|:green_apple:|

**Pull request merged**